Gems/ScriptCanvas/Code/Editor/View/Widgets/NodePalette/NodePaletteModel.cpp

```diff
@@ -1193,6 +1193,17 @@ namespace ScriptCanvasEditor
             senderInformation->m_displayName = details.m_name.empty() ? eventName : details.m_name.c_str();
             senderInformation->m_toolTip = details.m_tooltip.empty() ? "" : details.m_tooltip;
 
+            auto safeRegister = [](AZ::BehaviorMethod* method)
+            {
+                if (method && AZ::MethodReturnsAzEventByReferenceOrPointer(*method))
+                {
+                    const AZ::BehaviorParameter* resultParameter = method->GetResult();
+                    ScriptCanvas::ReflectEventTypeOnDemand(resultParameter->m_typeId, resultParameter->m_name, resultParameter->m_azRtti);
+                }
+            };
+
+            safeRegister(sender.m_event);
+            safeRegister(sender.m_broadcast);
```

Comment:
I saw this pattern in a different PR, makes me think this should be consolidated to avoid code management issues due to duplication